appendices/garbage-collection.md

```diff
@@ -1,35 +1,33 @@
 
-Pony-ORCA, yes _the killer whale_, is based on ideas from ownership and deferred, distributed, weighted **reference counting**. It adapts messaging systems of actors to keep the reference count consistent. The main challenges in concurrent garbage collection is the detection of cycles of sleeping actors in the actors graph, in the presence of concurrent mutation of this graph. With message passing you get deferred direct reference counting, a dedicated actor for the detection of (cyclic) garbage, and a confirmation protocol (to deal with the mutation of the actor graph).
+Pony-ORCA, yes _the killer whale_, is based on ideas from ownership and deferred, distributed, weighted **reference counting**. It adapts messaging systems of actors to keep the reference count consistent. The main challenges in concurrent garbage collection are the detection of cycles of sleeping actors in the actor's graph, in the presence of the concurrent mutation of this graph. With message passing, you get deferred direct reference counting, a dedicated actor for the detection of (cyclic) garbage, and a confirmation protocol (to deal with the mutation of the actor graph).
 
 1. _Soundness_: the technique collects only dead actors.
 
 2. _Completeness_: the technique collects all dead actors eventually.
 
-3. _Concurrency_: the technique does not require a stop-the-world step, clocks, timestamps, versioning, thread coordination, actor introspection, shared memory, read/write barriers or cache coherency.
+3. _Concurrency_: the technique does not require a stop-the-world step, clocks, time stamps, versioning, thread coordination, actor introspection, shared memory, read/write barriers or cache coherency.
 
 The type system ensures at compile time that your program can **never have data races**. It's **deadlock free**... Because Pony has **no locks**!
 
-When an actor has completed local execution and has no pending messages on its queue, it is _blocked_. An actor is _dead_, if it is blocked and all actors that have a reference to it are blocked, transitively. Collection of dead actors depends on being able to collect closed cycles of blocked actors.
+When an actor has completed local execution and has no pending messages on its queue, it is _blocked_. An actor is _dead_, if it is blocked and all actors that have a reference to it are blocked, transitively. A collection of dead actors depends on being able to collect closed cycles of blocked actors.
 
-The Pony type system guarantees soundness and race and dead-lock free concurrency by adhering to the following principles:
+The Pony type system guarantees soundness and race and deadlock free concurrency by adhering to the following principles:
```

Comment:
This should be changed to "The Pony type system guarantees race and deadlock free concurrency and soundness by adhering to the following principles:". This is implying that it's a soundness-free type system